backend/controller/evaljob_controller.go

```diff
@@ -0,0 +1,568 @@
+		}
+	}
+	return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+}
+
+func (r *EvalJobReconciler) getPod(ctx context.Context, job *lmevalservicev1beta1.EvalJob) (*corev1.Pod, error) {
+	var pod = corev1.Pod{}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: job.Namespace, Name: job.Name}, &pod); err != nil {
+		return nil, err
+	}
+	for _, ref := range pod.OwnerReferences {
+		if ref.APIVersion == job.APIVersion &&
+			ref.Kind == job.Kind &&
+			ref.Name == job.Name {
+
+			return &pod, nil
+		}
+	}
+	return nil, fmt.Errorf("pod doesn't have proper entry in the OwnerReferences")
+}
+
+func (r *EvalJobReconciler) deleteJobPod(ctx context.Context, job *lmevalservicev1beta1.EvalJob) error {
+	pod := corev1.Pod{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      job.Status.PodName,
+			Namespace: job.Namespace,
+			OwnerReferences: []v1.OwnerReference{
+				{
+					APIVersion: job.APIVersion,
+					Kind:       job.Kind,
+					Name:       job.Name,
+				},
+			},
+		},
+	}
+	return r.Delete(ctx, &pod, &client.DeleteOptions{})
+}
+
+func (r *EvalJobReconciler) handleComplete(ctx context.Context, log logr.Logger, job *lmevalservicev1beta1.EvalJob) (ctrl.Result, error) {
+	if job.Status.CompleteTime == nil {
+		r.Recorder.Event(job, "Normal", "JobCompleted",
+			fmt.Sprintf("Tthe EvalJob %s in namespace %s has completed",
+				job.Name,
+				job.Namespace))
+		// TODO: final wrap up/clean up
+		current := v1.Now()
+		job.Status.CompleteTime = &current
+		if err := r.Status().Update(ctx, job); err != nil {
+			log.Error(err, "failed to update status for completion")
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *EvalJobReconciler) handleCancel(ctx context.Context, log logr.Logger, job *lmevalservicev1beta1.EvalJob) (ctrl.Result, error) {
+	// delete the pod and update the state to complete
+	if _, err := r.getPod(ctx, job); err != nil {
+		// pod is gone. update status
+		job.Status.State = lmevalservicev1beta1.CompleteJobState
+		job.Status.Reason = lmevalservicev1beta1.FailedReason
+		job.Status.Message = err.Error()
+	} else {
+		job.Status.State = lmevalservicev1beta1.CompleteJobState
+		job.Status.Reason = lmevalservicev1beta1.CancelledReason
+		if err := r.deleteJobPod(ctx, job); err != nil {
+			// leave the state as is and retry again
+			log.Error(err, "failed to delete pod. scheduled a retry after 10 seconds")
+			return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, err
+		}
+	}
+
+	err := r.Status().Update(ctx, job)
+	if err != nil {
+		log.Error(err, "failed to update status for cancellation")
+	}
+	r.Recorder.Event(job, "Normal", "Cancelled",
+		fmt.Sprintf("Tthe EvalJob %s in namespace %s has cancelled and changed its state to Complete",
+			job.Name,
+			job.Namespace))
+	return ctrl.Result{}, err
+}
+
+func (r *EvalJobReconciler) createPod(job *lmevalservicev1beta1.EvalJob) *corev1.Pod {
+	var allowPrivilegeEscalation = false
```

Comment:
Interesting. I think it should be safe to always have these settings set, right? We certainly want to _allow_ these settings to be set, but we might want to make sure that it's possible for users to do otherwise. I could imagine the case where a user has a custom image with custom `Task` that does things like use temp files and read from disk. While this would certainly be an anti-pattern for a production-grade eval task, I wouldn't be at all surprised if users still wanted to be able to do it.

Reply:
I wolud say this kind of images will hit the SCC on OpenShift and fail. I hope we don't need to support that use case.